static/js/settings.js

```diff
@@ -124,6 +124,17 @@ function render_bots() {
     });
 }
 
+exports.generate_zuliprc_uri = function (email, api_key) {
+    var data = encodeURIComponent(
```

Comment:
It would be nice to extract a separate function here that **just** builds the contents of the file here, without adding in the complexity of serving up as an octet-stream.  Then generate_zuliprc_uri() would called to that new function and have one really simply job as well.

Separating out two functions would allow us to write simpler tests about just the contents of the file.  It's also possible that we'll eventually just want an option where we paste the file contents to the clipboard or simply display them in the browser, in light of the fact that browsers have limitations downloading files, or maybe you just want to paste the config stuff into your text editor.